src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -115,4 +122,49 @@ public void validate(@Nullable Schema inputSchema) {
       }
     }
   }
+
+  private void validatePartitionProperties(@Nullable Schema schema) {
+    Table table = BigQueryUtil.getBigQueryTable(getProject(), getDataset(), getTable(), getServiceAccountFilePath());
+    if (table != null) {
+      StandardTableDefinition tableDefinition = table.getDefinition();
+      TimePartitioning timePartitioning = tableDefinition.getTimePartitioning();
+      if (timePartitioning == null && createPartitionedTable != null && createPartitionedTable) {
```

Comment:
We documented that this property is ignored if the table already exists, so we should not through an exception here. It does seem like a weird configuration though, so it may make sense to log a warning. Something like:

The plugin is configured to auto-create a partitioned table, but table '%s' already exists without partitioning. Please verify the partitioning configuration.

Reply:
Done.